sound/soc/intel/common/soc-acpi-intel-tgl-match.c

```diff
@@ -96,6 +101,14 @@ struct snd_soc_acpi_mach snd_soc_acpi_intel_tgl_machines[] = {
 		.sof_fw_filename = "sof-tgl.ri",
 		.sof_tplg_filename = "sof-tgl-max98357a-rt5682.tplg",
 	},
+	{
+		.id = "MX98373",
+		.drv_name = "tgl_max98373_rt5682",
+		.machine_quirk = snd_soc_acpi_codec_list,
+		.quirk_data = &tgl_hp_5682_codec,
+		.sof_fw_filename = "sof-tgl.ri",
+		.sof_tplg_filename = "sof-tgl-max98373-rt5682.tplg",
```

Comment:
somehow gitHub lost my comment.

I think we should flip the logic and use rt5682 as a primary key and the amp as a quirk

Reply:
updated now.